modules/consentManagementUsp.js

```diff
@@ -0,0 +1,291 @@
+ * interacts with supported USP Consent APIs to grab the user's consent
+ * information and make it available for any USP (CCPA) supported adapters to
+ * read/pass this information to their system.
+ */
+import * as utils from '../src/utils';
+import { config } from '../src/config';
+import { uspDataHandler } from '../src/adapterManager';
+
+const DEFAULT_CONSENT_API = 'iab';
+const DEFAULT_CONSENT_TIMEOUT = 50;
+const USPAPI_VERSION = 1;
+
+export let consentAPI;
+export let consentTimeout;
+
+let consentData;
+let addedConsentHook = false;
+
+// consent APIs
+const uspCallMap = {
+  'iab': lookupUspConsent
+};
+
+/**
+ * This function handles interacting with an USP compliant consent manager to obtain the consent information of the user.
+ * Given the async nature of the USP's API, we pass in acting success/error callback functions to exit this function
+ * based on the appropriate result.
+ * @param {function(string)} uspSuccess acts as a success callback when USPAPI returns a value; pass along consentObject (string) from UPSAPI
+ * @param {function(string)} uspError acts as an error callback while interacting with USPAPI; pass along an error message (string)
+ * @param {object} hookConfig contains module related variables (see comment in requestBidsHook function)
+ */
+function lookupUspConsent(uspSuccess, uspError, hookConfig) {
+  function handleUspApiResponseCallbacks() {
+    const uspResponse = {};
+
+    function afterEach() {
+      if (uspResponse.usPrivacy) {
+        uspSuccess(uspResponse, hookConfig);
+      } else {
+        uspError('Unable to get USP consent string.', hookConfig);
+      }
+    }
+
+    return {
+      consentDataCallback: (consentResponse, success) => {
+        if (success && consentResponse.uspString) {
+          uspResponse.usPrivacy = consentResponse.uspString;
+        }
+        afterEach();
+      }
+    };
+  }
+
+  let callbackHandler = handleUspApiResponseCallbacks();
+  let uspapiCallbacks = {};
+
+  // to collect the consent information from the user, we perform a call to USPAPI
+  // to collect the user's consent choices represented as a string (via getUSPData)
+
+  // the following code also determines where the USPAPI is located and uses the proper workflow to communicate with it:
+  // - use the USPAPI locator code to see if USP's located in the current window or an ancestor window. This works in friendly or cross domain iframes
+  // - if USPAPI is not found, the iframe function will call the uspError exit callback to abort the rest of the USPAPI workflow
+  // - try to call the __uspapi() function directly, otherwise use the postMessage() api
+
+  // find the CMP frame/window
+  let f = window;
+  let uspapiFrame;
+  while (!uspapiFrame) {
+    try {
+      if (f.frames['__uspapiLocator']) uspapiFrame = f;
+    } catch (e) { }
+    if (f === window.top) break;
+    f = f.parent;
+  }
+
+  if (!uspapiFrame) {
+    return uspError('CMP not found.', hookConfig);
```

Comment:
Could we perhaps clarify this message to denote it was trying to find a USP CMP?